spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/transaction/jta/NarayanaJtaConfiguration.java

```diff
@@ -114,10 +115,12 @@ public TransactionManager narayanaTransactionManager() {
 	@Bean
 	@DependsOn("narayanaConfiguration")
 	public RecoveryManagerService narayanaRecoveryManagerService() {
+		RecoveryManager.delayRecoveryManagerThread();
```

Comment:
> Without it recovery manager is started automatically when any transaction logic is executed for the first time

Wouldn't that be a good thing? It would address my earlier concern about recovery perhaps being performed too late. But I'm confused about that now as those changes seem to have disappeared.